libraries/AP_DroneCAN/AP_DroneCAN.cpp

```diff
@@ -706,11 +706,16 @@ int16_t AP_DroneCAN::scale_esc_output(uint8_t idx){
     static const int16_t cmd_max = ((1<<13)-1);
     float scaled = 0;
 
+    scaled = hal.rcout->scale_esc_to_unity(_SRV_conf[idx].pulse);
+    // Prevent invalid values (from misconfigured scaling parameters) from sending non-zero commands
```

Comment:
```suggestion
    const float scaled = hal.rcout->scale_esc_to_unity(_SRV_conf[idx].pulse);
    // Prevent invalid values (from misconfigured scaling parameters) from sending non-zero commands
```

Reply:
Strictly I think we also need to constrain to -1 to 1. Maybe that would be better done in the `scale_esc_to_unity` method.

Reply:
You're right. Only the non-reversed servo channels got any constraint applied. I have fixed that now by moving the contrain_float up out of the if block.

It would be nice to do it in `scale_esc_to_unity`, but we don't know inside that function whether "off" is 0.0 or -1.0, and no way to signal to the caller that something is wrong and they should not spin anything.

It's not the best solution, and you're absolutely right that we should eliminate this at some point, but I'm restricting myself to the smallest change possible to fix the bug.